### PR TITLE
Add api fetch requests to get desired data

### PR DESCRIPTION
**Pull request by Caruki**
Create new routes and models to get desired api data

### CODE REVIEW

File: lib/models/shows.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        key: show.id,
+        title: show.name,
+        poster: `https://image.tmdb.org/t/p/w154/${show.poster_path}`,
+        rating: Math.floor(show.vote_average),
+      };
+      return trendingShowListView;
+    });
+    return trendingShowsListView;
+  } catch (error) {
+    throw new Error(error.message);
```
Comment (lmachens) [inline lib/models/shows.js:21]: If you just rethrow the error, you can remove the try-catch here. (happens multiple times)

File: lib/models/shows.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    '-' +
+    currentDate.getDate();
+  const airDateGTE =
+    currentDate.getFullYear() +
+    '-0' +
+    (currentDate.getMonth() - 1) +
+    '-' +
+    currentDate.getDate();
+  try {
+    console.log(airDateGTE);
```
Comment (lmachens) [inline lib/models/shows.js:41]:
```suggestion
```

File: lib/models/shows.js [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+      fetch(`
+    https://api.themoviedb.org/3/tv/${showId}?api_key=${process.env.TMDB_API_KEY}&append_to_response=images`),
+      fetch(`
+    https://api.themoviedb.org/3/tv/${showId}/credits?api_key=${process.env.TMDB_API_KEY}`),
+      fetch(`
+    https://api.themoviedb.org/3/tv/${showId}/videos?api_key=${process.env.TMDB_API_KEY}`),
+    ]);
+    const allShowDetails = await Promise.all(
+      responses.map((response) => response.json())
+    );
```
Comment (lmachens) [inline lib/models/shows.js:74]: I think you can remove the first `await Promise.all`

```suggestion
    const responses = [
      fetch(`
    https://api.themoviedb.org/3/tv/${showId}?api_key=${process.env.TMDB_API_KEY}&append_to_response=images`),
      fetch(`
    https://api.themoviedb.org/3/tv/${showId}/credits?api_key=${process.env.TMDB_API_KEY}`),
      fetch(`
    https://api.themoviedb.org/3/tv/${showId}/videos?api_key=${process.env.TMDB_API_KEY}`),
    ];
    const allShowDetails = await Promise.all(
      responses.map((response) => response.json())
    );
```

File: lib/models/shows.js [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+  try {
+    const responses = await Promise.all([
+      fetch(`
+    https://api.themoviedb.org/3/tv/${showId}?api_key=${process.env.TMDB_API_KEY}&append_to_response=images`),
+      fetch(`
+    https://api.themoviedb.org/3/tv/${showId}/credits?api_key=${process.env.TMDB_API_KEY}`),
+      fetch(`
+    https://api.themoviedb.org/3/tv/${showId}/videos?api_key=${process.env.TMDB_API_KEY}`),
+    ]);
+    const allShowDetails = await Promise.all(
```
Comment (lmachens) [inline lib/models/shows.js:72]: The later usage of `allShowDetails[0]` is not easy to understand.
I would give them better names:

```suggestion
    const [show, credits, videos] = await Promise.all(
```

File: lib/models/shows.js (hunk 3 above)
Comment (Caruki) [reply lib/models/shows.js:74]: I tried it and it actually marks an error: that response.json() is not a function...

**Pull request merged**